lib/_sendEvent.ts

```diff
@@ -1,66 +1,68 @@
-import { ClickReport } from './click';
-import { ConversionReport } from './conversion';
-import { SearchReport } from './search';
+import { InsightsSearchClickEvent } from './click';
+import { InsightsSearchConversionEvent } from './conversion';
 
-declare var process : {
+declare var process: {
   env: {
-    NODE_ENV: string
-  }
-}
+    NODE_ENV: string;
+  };
+};
 
-export type AnalyticsEvent = {
+export type InsightsEventTypes = 'click' | 'conversion';
+export type InsightsEvent = {
+  eventType: InsightsEventTypes;
+  eventName: string;
   userID: string;
   timestamp: number;
   indexName: string;
-}
-
-export type ReportEvent = 'search'|'click'|'conversion'
+};
 
 /**
  *  Sends data to endpoint
- * @param eventType 'click'|'conversion'
- * @param eventData ConversionReport|ClickReport
+ * @param eventType InsightsEventTypes
+ * @param eventData InsightsSearchClickEvent|InsightsSearchConversionEvent
  */
-export function sendEvent(eventType: ReportEvent, eventData: ClickReport | ConversionReport | SearchReport) {
-
+export function sendEvent(
+  eventType: InsightsEventTypes,
+  eventData: InsightsSearchClickEvent | InsightsSearchConversionEvent
+) {
   // Add client timestamp and userID
-  eventData.timestamp = Date.now()
+  eventData.timestamp = Date.now();
   eventData.userID = this._userID;
+  eventData.eventType = eventType;
+  // TODO: check eventType is matching eventData
```

Comment:
When would it need to be done?

Reply:
in the PR implementing click/conversion/view signatures